core/src/core_tests/workflow_tasks.rs

```diff
@@ -1687,54 +1694,6 @@ async fn tasks_from_completion_are_delivered() {
     core.shutdown().await;
 }
 
-#[tokio::test]
-async fn evict_missing_wf_during_poll_doesnt_eat_permit() {
```

Comment:
This test serves little purpose now since all the permits are owned. It's very high-effort to make work since the poll acquires the permit internally, thus a permit "used" immediately again before we get a new task. In this test, there is no new task, which means we shut down (which releases the permit, but also renders the check pointless).